### PR TITLE
RN: Fix passing non-link tokens without public_key

### PR DESCRIPTION
**Pull request by amytang0**

### CODE REVIEW

File: ios/RNLinksdk.m [hunk 1]
```diff
@@ -99,14 +99,16 @@ - (void)stopObserving {
     NSString *paymentTokenInput = [RCTConvert NSString:configuration[kRNLinkKitConfigPaymentTokenKey]];
     NSString *oauthStateId = [RCTConvert NSString:configuration[kRNLinkKitConfigOAuthStateIdKey]];
     NSString *institution = [RCTConvert NSString:configuration[kRNLinkKitConfigInstitutionKey]];
+    NSString *publicKey = [RCTConvert NSString:configuration[kRNLinkKitConfigInstitutionKey]];
```
Comment (wkiefer) [inline ios/RNLinksdk.m:102]: Should be `kRNLinkKitConfigPublicKeyKey` instead of `kRNLinkKitConfigInstitutionKey`

File: ios/RNLinksdk.m [hunk 2]
```diff
@@ -99,14 +99,16 @@ - (void)stopObserving {
...
 
     PLKConfiguration* linkConfiguration;
     BOOL isUsingLinkToken = [linkTokenInput length] > 0 && [linkTokenInput hasPrefix:kRNLinkKitLinkTokenPrefix];
+    BOOL isUsingPublicKey = kRNLinkKitConfigPublicKeyKey;
```
Comment (wkiefer) [inline ios/RNLinksdk.m:106]: this assignment is always going to be true. this should be checking if `publicKey` has length instead

File: ios/RNLinksdk.m (hunk 1 above)
Comment (amytang0) [reply ios/RNLinksdk.m]: My bad, thank you!

File: ios/RNLinksdk.m [hunk 3]
```diff
@@ -99,14 +99,16 @@ - (void)stopObserving {
...
 
     PLKConfiguration* linkConfiguration;
     BOOL isUsingLinkToken = [linkTokenInput length] > 0 && [linkTokenInput hasPrefix:kRNLinkKitLinkTokenPrefix];
+    BOOL isUsingPublicKey = [publicKey length] > 0;
 
-    if (isUsingLinkToken) {
-      linkConfiguration = [self getLinkTokenConfiguration: configuration];
-    } else {
+    if (isUsingPublicKey) {
       linkConfiguration = [self getLegacyLinkConfiguration: configuration];
```
Comment (afh) [inline ios/RNLinksdk.m:109]: 💅  I suggest to name this `getLinkPublicKeyConfiguration`; could be done in a separate PR.

**Review by afh — APPROVED** [review]
Thanks for making these changes @amytang0.
Left a minor unrelated nit (_v.s._), which we may want to address in a separate PR.

**Comment by amytang0** [comment]
Leaving `getLinkPublicKeyConfiguration` for later when we update the SDK version

**Pull request merged**